pages/[resourceType]/[id]/care-gaps.tsx

```diff
@@ -57,6 +61,116 @@ const CareGapsPage = () => {
     }
   }, [radioValue]);
 
+  /**
+   * createRequestPreview builds the request preview with the care-gaps state variables
+   * @returns the request preview as a string
+   */
+  const createRequestPreview = () => {
+    //dates are formatted to be in the form "YYYY-MM-DD", with no timezone info
+    let requestPreview = `/Measure/$care-gaps?measureId=${id}&periodStart=${DateTime.fromISO(
+      periodStart.toISOString(),
+    ).toISODate()}&periodEnd=${DateTime.fromISO(
+      periodEnd.toISOString(),
+    ).toISODate()}&status=open-gap`;
+    if (radioValue) {
+      if (radioValue.toLowerCase() === "subject" && patientValue) {
+        requestPreview += `&subject=${patientValue}`;
+      } else if (radioValue.toLowerCase() === "organization" && organizationValue) {
+        requestPreview += `&organization=${organizationValue}`;
+        practitionerValue
+          ? (requestPreview += `&practitioner=${practitionerValue}`)
+          : requestPreview;
+      }
+      programValue ? (requestPreview += `&program=${programValue}`) : requestPreview;
+    }
+
+    return requestPreview;
+  };
+
+  //true if user has selected/entered a correct combination of inputs
+  const validSelections = () => {
+    if (
+      (periodStart && periodEnd && radioValue === "Subject" && patientValue) ||
+      (periodStart && periodEnd && radioValue === "Organization" && organizationValue)
+    ) {
+      return true;
+    } else return false;
+  };
+
+  //handles sending the care-gaps request and processes the response
+  function calculateCareGapsHandler() {
+    let customMessage = <Text weight={500}>Problem connecting to server:&nbsp;</Text>;
+    let notifProps: NotificationProps = {
+      message: customMessage,
+      color: "red",
+      icon: <X size={18} />,
+      autoClose: false,
+    };
+    let fetchStatus = { status: 500, statusText: "Failed fetch request" };
+    setLoadingRequest(true);
+
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}${createRequestPreview()}`)
+      .then((response) => {
+        fetchStatus = { status: response.status, statusText: response.statusText };
+        return response.json();
```

Comment:
Remember to use strong typing/type inference. In fetch requests where we know the return type for certain. In this case, response.json() is a `Promise<fhirJson.Parameters | fhirJson.OperationOutcome>`.

I think I made a similar comment in the `MeasuredatePickers` component, so you can look back at that for guidance